akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -44,6 +44,16 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
 
       def render(h: HttpHeader) = r ~~ h ~~ CrLf
 
+      val hostString: InetSocketAddress ⇒ String = {
+        // Retrieve the original host string that was given (IP or DNS name) if running Java 7 or later
+        // using the getHostString() method. This avoids a reverse DNS query from calling getHostName()
+        // if the original host string is an IP address.
+        val clazz = classOf[InetSocketAddress]
+        address ⇒ clazz.getMethods.collectFirst {
+          case m if m.getName == "getHostString" => m.invoke(address)
+        }.getOrElse(address.getHostName).asInstanceOf[String]
+      }
```

Comment:
Yeah, my bad. Also would it be better to move hostString val to one level up - from onPush to HttpRequestRenderer or even to HttpRequestRendererFactory? Just to execute this as less times as possible. 

Also it seems to me that ```require``` statements in try/catch will throw exception the same as ```m.invoke``` (if any) and this exception will be ignored as we'll switch to 1.6 ```getHostName```. So here ```require``` is more for visibility and documentation so I believe a comment with explanation why we are using reflection here should be put instead as it is not common for this project. Does it sound reasonable?